tgui/packages/tgui/interfaces/ChemiCompiler/index.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Garash2k (https://github.com/Garash2k)
+ * @license ISC
+ */
+
+import { Window } from '../../layouts';
+import { useBackend } from '../../backend';
+import { ChemiCompilerData } from './type';
+import { ChemiCompilerTextArea } from './ChemiCompilerTextArea';
+import { ChemiCompilerReservoirs } from './ChemiCompilerReservoirs';
+import { ChemiCompilerMemory } from './ChemiCompilerMemory';
+import { Stack } from '../../components';
+
+export const ChemiCompiler = (_props, context) => {
+  const { data } = useBackend<ChemiCompilerData>(context);
+  const { theme } = data;
+  return (
+    <Window width={600} height={500} theme={theme}>
+      <Window.Content>
+        <Stack fill>
+          <Stack.Item grow>
+            <ChemiCompilerTextArea />
+          </Stack.Item>
+          <Stack.Item basis={18} textAlign="center">
```

Comment:
Nice clean layout. 👍 

Only suggestion here is to move `basis={18}` to `basis={SIDEBAR_WIDTH}` or whatever, then define that above. Magic numbers bad.